src/test/java/org/codehaus/plexus/archiver/zip/ZipArchiverTest.java

```diff
@@ -809,6 +810,10 @@ public void testZipNonConcurrentResourceCollection()
     {
         final File tarFile = getTestFile( "target/output/zip-non-concurrent.tar" );
         TarArchiver tarArchiver = (TarArchiver) lookup( Archiver.class, "tar" );
+        // Override uId and gId - in standard mode on OS where uId or gId can have long values creation of tar can fail
+        // We can not use posix mode because mTime can have nanoseconds in posix but zip doesn't so assertions can fail
+        tarArchiver.setOverrideUid(100);
```

Comment:
This tests the zip file creation from resource collection so not getting the uid/gui from the filesystem should not affect what is asserted. But I wonder how tar tests are not failing then. They should test if the uid is correct.

Reply:
In tar tests posix mode is used.

But here also timestamp is compared. Looks like in zip nanoseconds are dropped.